packit/utils/repo.py

```diff
@@ -20,6 +20,52 @@
 logger = logging.getLogger(__name__)
 
 
+class RepositoryCache:
+    def __init__(self, cache_path: Path, add_new=False) -> None:
+        self.cache_path = cache_path
+        self.add_new = add_new
+
+    @property
+    def cached_projects(self) -> List[str]:
+        if not self.cache_path.is_dir():
+            self.cache_path.mkdir(parents=True)
+        return [f.name for f in self.cache_path.iterdir() if f.is_dir()]
+
+    def __contains__(self, project):
+        return project in self.cached_projects
+
+    def __getitem__(self, item):
+        if item in self.cached_projects:
+            return self.cache_path / item
+        return None
+
+    def get_repo(
+        self, url: str, directory: Union[Path, str] = None, project_name=None
+    ) -> git.Repo:
+        directory = str(directory) if directory else tempfile.mkdtemp()
+
+        if is_git_repo(directory=directory):
+            logger.debug(f"Repo already exists in {directory}.")
+            return git.repo.Repo(directory)
+
+        logger.debug(f"Cloning repo {url} -> {directory}")
+
+        reference_repo = self.cache_path.joinpath(project_name)
+        if project_name not in self.cached_projects and self.add_new:
```

Comment:
we should also figure out how to update those cached repositories

Reply:
Currently, this is only to be able to use that from service so we can do that separately as a cron job.

Since we can clone the repo during the parsing of CLI arguments, it's not easy to support CLI for now. (I am planning to finally do something with `LocalProject` to allow this.)